src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -182,6 +182,59 @@ public static bool IsWhiteSpace(this ReadOnlySpan<char> span)
             return true;
         }
 
+        // TODO: Is this optimization required?
+        private static readonly Type s_byte = typeof(byte);
```

Comment:
Is this optimization necessary?

Reply:
This is not an optimization. It is de-optimization.

The JIT recognizes ` if (typeof(T) == typeof(byte))` pattern and evaluates into constant. It is not able to do it once you cache the type in a static variable.